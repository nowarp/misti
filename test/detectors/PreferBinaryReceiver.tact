message(1) One {}

contract Example {
    receive("one") { } // Bad
    external("two") { } // Bad

    receive(_: One) { } // OK
    receive(_: String) { } // OK
    receive() { } // OK
    external(_: One) { } // OK
    external(_: String) { } // OK
    external() { } // OK
    bounced(_: Slice) { } // OK
}
