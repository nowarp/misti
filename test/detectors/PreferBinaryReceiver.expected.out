[LOW] PreferBinaryReceiver: Prefer binary receiver
test/detectors/PreferBinaryReceiver.tact:4:5:
  3 | contract Example {
> 4 |     receive("one") { } // Bad
          ^~~~~~~~~~~~~~~~~~
  5 |     external("two") { } // Bad
Using text receivers is a gas-expensive operation
Help: Consider changing it to a binary receiver
See: https://nowarp.io/tools/misti/docs/detectors/PreferBinaryReceiver

[LOW] PreferBinaryReceiver: Prefer binary receiver
test/detectors/PreferBinaryReceiver.tact:5:5:
  4 |     receive("one") { } // Bad
> 5 |     external("two") { } // Bad
          ^~~~~~~~~~~~~~~~~~~
  6 | 
Using text receivers is a gas-expensive operation
Help: Consider changing it to a binary receiver
See: https://nowarp.io/tools/misti/docs/detectors/PreferBinaryReceiver